components/Cards/src/MDCCard.m

```diff
@@ -0,0 +1,238 @@
+static NSString *const MDCCardShadowColorsKey = @"MDCCardShadowColorsKey";
+static NSString *const MDCCardBorderWidthsKey = @"MDCCardBorderWidthsKey";
+static NSString *const MDCCardBorderColorsKey = @"MDCCardBorderColorsKey";
+static NSString *const MDCCardInkViewKey = @"MDCCardInkViewKey";
+
+
+@implementation MDCCard {
+  NSMutableDictionary<NSNumber *, NSNumber *> *_shadowElevations;
+  NSMutableDictionary<NSNumber *, UIColor *> *_shadowColors;
+  NSMutableDictionary<NSNumber *, NSNumber *> *_borderWidths;
+  NSMutableDictionary<NSNumber *, UIColor *> *_borderColors;
+  CGPoint _lastTouch;
+}
+
++ (Class)layerClass {
+  return [MDCShadowLayer class];
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+  self = [super initWithCoder:coder];
+  if (self) {
+    _shadowElevations = [coder decodeObjectForKey:MDCCardShadowElevationsKey];
+    _shadowColors = [coder decodeObjectForKey:MDCCardShadowColorsKey];
+    _borderWidths = [coder decodeObjectForKey:MDCCardBorderWidthsKey];
+    _borderColors = [coder decodeObjectForKey:MDCCardBorderColorsKey];
+    _inkView = [coder decodeObjectForKey:MDCCardInkViewKey];
+    [self commonMDCCardInit];
+  }
+  return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder {
+  [super encodeWithCoder:coder];
+  [coder encodeObject:_shadowElevations forKey:MDCCardShadowElevationsKey];
+  [coder encodeObject:_shadowColors forKey:MDCCardShadowColorsKey];
+  [coder encodeObject:_borderWidths forKey:MDCCardBorderWidthsKey];
+  [coder encodeObject:_borderColors forKey:MDCCardBorderColorsKey];
+  [coder encodeObject:_inkView forKey:MDCCardInkViewKey];
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCCardInit];
+  }
+  return self;
+}
+
+- (void)commonMDCCardInit {
+  if (_inkView == nil) {
+    _inkView = [[MDCInkView alloc] initWithFrame:self.bounds];
+    _inkView.autoresizingMask = (UIViewAutoresizingFlexibleWidth |
+                                 UIViewAutoresizingFlexibleHeight);
+    _inkView.usesLegacyInkRipple = NO;
+    _inkView.layer.zPosition = FLT_MAX;
+    [self addSubview:self.inkView];
+  }
+
+  if (_shadowElevations == nil) {
+    _shadowElevations = [NSMutableDictionary dictionary];
+    _shadowElevations[@(UIControlStateNormal)] = @(1.f);
+    _shadowElevations[@(UIControlStateHighlighted)] = @(8.f);
+  }
+
+  if (_shadowColors == nil) {
+    _shadowColors = [NSMutableDictionary dictionary];
+    _shadowColors[@(UIControlStateNormal)] = [UIColor blackColor];
+  }
+
+  if (_borderColors == nil) {
+    _borderColors = [NSMutableDictionary dictionary];
+  }
+
+  if (_borderWidths == nil) {
+    _borderWidths = [NSMutableDictionary dictionary];
+    _borderWidths[@(UIControlStateNormal)] = @(0.f);
+  }
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  [self updateShadowElevation];
+  [self updateShadowColor];
+  [self updateBorderWidth];
+
+  self.cornerRadius = 4.f;
```

Comment:
is this the right place where this should go? Rob talked about the UI_APPEARANCE and not being able to set it in the init, but I always assumed its important to set these in init. Any ideas @ianegordon @randallli ?